### ISSUE TITLE
smooth-scroll-margin vs scroll-margin

### ISSUE THREAD

**Issue by agriffis** [syl20bnr/spacemacs#6097]
(I tried to report this within Spacemacs but it seems like [SPC h I] doesn't do anything from terminal emacs-client.)

The default setting for `smooth-scroll-margin` is 5, but the default for `scroll-margin` is 0. This means that when one presses `C-f` to scroll forward by a page, the cursor is at the very top of the window according to `scroll-margin`. Then pressing any key will adjust the scrolling for `smooth-scroll-margin`, which is a jarring experience every time.

Furthermore, I tried to work around this by setting `smooth-scroll-margin 0` but that seems to disable smooth scrolling entirely, causing emacs to revert to the recentering behavior.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.20
- Spacemacs branch: master (rev. 9c7250a)
- Graphic display: nil
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(auto-completion better-defaults git github
                 (clojure :variables clojure-enable-fancify-symbols t)
                 emacs-lisp html markdown org python django
                 (syntax-checking :variables syntax-checking-enable-tooltips nil)
                 themes-megapack)
```

**Comment by agriffis** [comment]
This is my current workaround:

```lisp
  (setq scroll-margin 1)
  (setq smooth-scroll-margin 1)
```

which avoids the jarring effect, and gets as close to 0 as allows smooth scrolling to still take effect.

**Comment by NJBS** [comment]
I'd also like to add that this behavior occurs with `evil-scroll-line-down` (`C-e`) and `evil-scroll-line-up` (`C-y`). `scroll-margin` stopped being set [in this commit](https://github.com/syl20bnr/spacemacs/commit/0ca7389#diff-1ddb0627a4b3866462d091cc62448701L248) in favor of `smooth-scroll-margin`, but it seems that not all scrolling methods respect `smooth-scroll-margin`.

**Comment by sooheon** [comment]
I've found I don't need to touch smooth-scroll-margin.

```
  (setq scroll-conservatively 101
        scroll-margin 3
        scroll-preserve-screen-position 't)
```

This will replicate evil scrolloff behavior pretty well in my experience.

**Comment by StreakyCobra** [comment]
> (I tried to report this within Spacemacs but it seems like [SPC h I] doesn't do anything from terminal emacs-client.)

https://github.com/syl20bnr/spacemacs/issues/5902 and  https://github.com/syl20bnr/spacemacs/pull/5912

**Comment by sooheon** [comment]
@agriffis try this:
```
  (setq scroll-conservatively 101
        scroll-margin 0
        scroll-preserve-screen-position 't)
```

**Comment by agriffis** [comment]
@sooheon that works for me, thanks

**Comment by syl20bnr** [comment]
So we don't need smooth-scrolling ?

**Comment by sooheon** [comment]
In my recent experience, no. Those three default variables will do it for me, and often better: for example, at least for me, the way it behaves after some screenwise movements like `C-f` or `G` is better. W/ smooth-scroll, the margin is sometimes not respected until you make a linewise movement.

**Comment by syl20bnr** [comment]
I tried without smooth-scrolling and the result was not satisfying, the point is jumping depending on the responsiveness which is not the case with smooth-scrolling.

I investigated to make smooth-scrolling to work with `smooth-scrolling-margin` to 0, turns out it is not trivial. The number of lines computed is not very reliable (especially for the bottom of the window) and word wrap is not easy to handle. smooth-scrolling tries to keep the point on screen at all costs, if it fails then the point jumps.

**Comment by syl20bnr** [comment]
Actually I retested and found that scroll-conservatively does the job, like @sooheon.
I pushed https://github.com/syl20bnr/spacemacs/issues/6097 to let people test in their config.

**Comment by TheBB** [comment]
Since nobody complained in nine days I'm marking this as fixed.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
